passes/source2il.nim

```diff
@@ -422,6 +422,18 @@ proc fitExpr(c; e: sink Expr, target: SemType): Expr =
     result = Expr(stmts: e.stmts, expr: @[Node(kind: IntVal)],
                   typ: errorType())
 
+proc fitExprStrict(c; e: sink Expr, typ: SemType): Expr =
```

Comment:
I guess the other option is we cast if it's a subtype, but this is probably for the best.